### PR TITLE
Moved CI to GitHub actions

### PR DESCRIPTION
**Pull request by nkern**
In line with hera_cal and other repos, this moves our CI to GitHub actions from travis.ci

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    name: Tests
+    env:
+      ENV_NAME: hera_pspec_tests
+      PYTHON: ${{ matrix.python-version }}
+      OS: ${{ matrix.os }}
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macos-latest]
+        python-version: [3.6, 3.7]
```
Comment (philbull) [inline .github/workflows/ci.yml:20]: Do we want to explicitly support Python 3.6? The final supported release was in August 2020, and it's security bugfixes only from now on. I would be fine with moving to Python 3.7 only...

**Review by philbull — APPROVED** [review]
Looks good to me!

File: .github/workflows/ci.yml (hunk 1 above)
Comment (nkern) [reply .github/workflows/ci.yml:20]: Hmm interesting point. Perhaps its good to include it here until it is officially deprecated? It seems to me a lot of people use Python 3.6, so maybe its good to just keep it in our tests until we have reason to suspect people will be forced to move on?

Comment (nkern) [reply .github/workflows/ci.yml:20]: I'm going to merge the PR, but if you think we should address this please raise an issue, thanks!

**Pull request merged**